### PR TITLE
jami: 20211223.2.37be4c3 ->  20220503.1550.0f35faa

### PR DESCRIPTION
**Pull request by linsui**
###### Description of changes

switch to qt6, depends on https://github.com/NixOS/nixpkgs/pull/175306

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Need to be fixed:

1. wmclass is `net.jami.` which should be `net.jami.Jami`
2. crash with `QT_QPA_PLATFORM=xcb`

### CODE REVIEW

**Comment by NickCao** [comment]
Looked around in src/3rdparty/chromium/ui/events/ozone/layout/xkb/xkb_keyboard_layout_engine.cc, and found that the xkb config path is unfortunately hardcoded, no wonder why setting QT_XKB_CONFIG_ROOT did not work.

**Comment by NickCao** [comment]
```
diff --git a/pkgs/development/libraries/qt-6/modules/qtwebengine.nix b/pkgs/development/libraries/qt-6/modules/qtwebengine.nix
index 88254ab1589..5dea9556866 100644
--- a/pkgs/development/libraries/qt-6/modules/qtwebengine.nix
+++ b/pkgs/development/libraries/qt-6/modules/qtwebengine.nix
@@ -64,6 +64,7 @@
 , lcms2
 , re2
 , kerberos
+, xkeyboard_config
 , enableProprietaryCodecs ? true
 }:

@@ -127,6 +128,9 @@ qtModule rec {

     sed -i -e '/libpci_loader.*Load/s!"\(libpci\.so\)!"${pciutils}/lib/\1!' \
       src/3rdparty/chromium/gpu/config/gpu_info_collector_linux.cc
+
+    substituteInPlace src/3rdparty/chromium/ui/events/ozone/layout/xkb/xkb_keyboard_layout_engine.cc \
+      --replace "/usr/share/X11/xkb" "${xkeyboard_config}/share/X11/xkb"
   '';

   cmakeFlags = [
```

File: pkgs/applications/networking/instant-messengers/jami/daemon.nix [hunk 1]
```diff
@@ -58,16 +58,22 @@ let
...
+        owner = "savoirfairelinux";
+        repo = "pjproject";
+        rev = version;
+        sha256 = "sha256-6t+3b7pvvwi+VD05vxtujabEJmWmJTAeyD/Dapav10Y=";
+      };
 
       patches = old.patches ++ (map (x: patch-src + x) (readLinesToList ./config/pjsip_patches));
+
+      configureFlags = (readLinesToList ./config/pjsip_args_common)
+        ++ lib.optionals stdenv.isLinux (readLinesToList ./config/pjsip_args_linux);
```
Comment (NickCao) [inline pkgs/applications/networking/instant-messengers/jami/daemon.nix:76]: Why use separated files for these flags, they don't seem particularly length.

Comment (linsui) [reply pkgs/applications/networking/instant-messengers/jami/daemon.nix:76]: This is mainly for the update scripts.

**Comment by linsui** [comment]
@NickCao Will you fix the xkb problem along with the qt.conf?

**Comment by NickCao** [comment]
I'd like to, but given that exams are approaching, no eta.

**Comment by NickCao** [comment]
According to https://doc.qt.io/qt-6/qt-conf.html, these paths can only be specified once? What if we need to load something from multiple paths, guess qt-env still has to be a thing.

**Comment by milahu** [comment]
> @milahu Would you like to implement the fix you proposed in [#169296 (comment)](https://github.com/NixOS/nixpkgs/pull/169296#issuecomment-1133896656) to ease futures works?

im on it

> What if we need to load something from multiple paths, guess qt-env still has to be a thing.

yes. qtconfig returns strings, and these are used directly as file paths
we would need some kind of lazy-evaluating string proxy
which resolves the actual file path as late as possible.
probably this can be done in C++ but ... expensive to implement

File: pkgs/applications/networking/instant-messengers/jami/default.nix [hunk 2]
```diff
@@ -4,21 +4,21 @@
...
 }:
 
 rec {
-  version = "20211223.2.37be4c3";
+  version = "20220503.1550.0f35faa";
 
   src = fetchzip {
     url = "https://dl.jami.net/release/tarballs/jami_${version}.tar.gz";
-    sha256 = "1zw9azwmxr4991nq5kl527lbwlj7psrissgvrkl1kxxbfbdncbhh";
+    sha256 = "iCmsgjgGogNjj1k0sYRqx59ZEwFZcJOeVGBNyBlcy1M=";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/instant-messengers/jami/default.nix:15]: Sri hashes must retain their prefix.

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -35380,12 +35380,12 @@ with pkgs;
 
   btcdeb = callPackage ../applications/blockchains/btcdeb { };
 
-  inherit (callPackage ../applications/networking/instant-messengers/jami {
+  jami = callPackage ../applications/networking/instant-messengers/jami {
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:35383]: Looks to me like we want to use callPackages

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**